src/test/scala/pokerData/HandGenerators.scala

```diff
@@ -246,14 +244,43 @@ object HandGenerators {
     rank <- genRank
     grouped: Map[Rank, List[Card]] = Deck.all.groupBy(_.rank)
     pair <- pick(2, grouped(rank))
-    //    rankingList = HandRank.all.filterNot(_ == Pair)
-    rest <- pick(5, Deck.all.filterNot(pair.contains(_))).retryUntil(x =>
-      Hand.rank(pair.toList ++ x.toList) match {
-        case _: Pair => true
-        case _       => false
-      }
+
+    deck = Deck.all.filterNot(_.rank == rank)
+    optHand: List[Option[Card]] = List(
+      Some(pair(0)),
+      Some(pair(1)),
+      none,
+      none,
+      none,
+      none,
+      none
     )
-  } yield Pair(rank, rest.sorted.reverse.toList)
+    (_, cards) = optHand.foldLeft((deck, List(pair(0).some, pair(1).some))) {
+      (s: (List[Card], List[Option[Card]]), v) =>
+        v match {
+          case Some(_) =>
+            s // Something is fishy... Like I'm missing a combinator or something..
+          case None =>
+            val (deck, hand) =
+              ensureNotTwoPairFlushStraight(s._1, s._2.sequence)
+            (deck, hand.sequence)
+        }
+    }
+    rest = cards.sequence.get // TODO Get???!!!! arrghhh
```

Comment:
FWIW there should be a way to lift `Option` to `Gen`

Reply:
ty!  I will investigate.